### PR TITLE
add autoupdating checking and notifying

### PR DESCRIPTION
**Pull request by david1alvarez**
### Motivation

We have the ability to check for updates automatically, but have only been doing so at the launch of the application. We want to be able to check semi-continuously for new updates to help users that leave their desktop wallet open stay up to date.

### In this PR

- linting updates to allow for variable declaration and usage to be out of strict order. All declarations get hoisted in JavaScript/Typescript, and there are certain times when its much cleaner to code this way.
- Change how we declare our route rendering logic to enable the use of react hooks. 
- Make use of said hooks to set up a background running task that checks for application updates once per hour
- Alter the code used for setting up and calling electron's `autoUpdater` functionality using method calls rather than importing and running lines from another file directly.
- Set up toast notifications that require user dismissal (see screenshot)

Addresses ticket https://app.asana.com/0/1200242791884109/1201822738255848/f

### Future Work

We need to review how our application is structured at a top level. The logic added in this PR to the `Routes` section of the tree shouldn't be there, but should be in some sort of application-container-level component.

### Screen Shots

Screen Shot 2022-03-01 at 6 33 55 PM

### CODE REVIEW

File: app/App.tsx [hunk 1]
```diff
@@ -43,15 +43,15 @@ const App: FC = () => {
       ipcRenderer.removeAllListeners('set-theme-light');
       ipcRenderer.removeAllListeners('set-theme-dark');
     };
-  });
+  }, []);
```
Comment (Shramp) [inline app/App.tsx:46]: good catch!

File: app/contexts/FullServiceContext.tsx [hunk 2]
```diff
@@ -110,14 +110,20 @@ const initialFullServiceState: FullServiceState = {
   addresses: { addressIds: [], addressMap: {} },
   contacts: [],
   encryptedPassphrase: undefined,
+  feePmob: '',
```
Comment (Shramp) [inline app/contexts/FullServiceContext.tsx:113]: what's the deal with these changes? they don't seem related to the updating

**Review by Shramp — APPROVED** [review]

**Comment by Shramp** [comment]
LGTM, just one question about the changes to initial app state

**Comment by david1alvarez** [comment]
@Shramp they're logically unrelated, but part of generally attempting to quell the million-or-so errors vscode is throwing
